package/MDAnalysis/analysis/msd.py

```diff
@@ -0,0 +1,419 @@
+                Molecular Dynamics [Article v1.0]. Living J. Comput. Mol. Sci. 
+                2019, 1 (1).
+
+.. [Yeh2004] Yeh, I. C.; Hummer, G. System-Size Dependence of Diffusion
+                Coefficients and Viscosities from Molecular Dynamics 
+                Simulations with Periodic Boundary Conditions. 
+                J. Phys. Chem. B 2004, 108 (40), 15873–15879.
+                
+.. [Bulow2020] von Bülow, S.; Bullerjahn, J. T.; Hummer, G. Systematic 
+                Errors in Diffusion Coefficients from Long-Time Molecular 
+                Dynamics Simulations at Constant Pressure. 2020. 
+                arXiv:2003.09205 [Cond-Mat, Physics:Physics].
+
+
+
+Classes and Functions
+---------------------
+
+.. autoclass:: EinsteinMSD
+    :members:
+
+"""
+
+from __future__ import division, absolute_import
+
+import numpy as np
+import logging
+from ..due import due, Doi
+from .base import AnalysisBase
+
+logger = logging.getLogger('MDAnalysis.analysis.msd')
+
+due.cite(Doi("10.21105/joss.00877"),
+         description="Mean Squared Displacements with tidynamics",
+         path="MDAnalysis.analysis.msd",
+         cite_module=True)
+due.cite(Doi("10.1051/sfn/201112010"),
+         description="FCA fast correlation algorithm",
+         path="MDAnalysis.analysis.msd",
+         cite_module=True)
+del Doi
+
+
+class EinsteinMSD(AnalysisBase):
+    r"""Class to calculate Mean Squared Displacement by the Einstein relation.
+
+    Attributes
+    ----------
+    dim_fac : int
+        Dimensionality :math:`d` of the MSD.
+    timeseries : :class:`numpy.ndarray`
+        The averaged MSD with respect to lag-time.
+    msd_per_particle : :class:`numpy.ndarray`
+        The MSD of each individual particle with respect to lag-time.
+    n_frames : int
+        Number of frames in trajectory.
+    n_particles : int
+        Number of particles MSD was calculated over.
+
+    """
+
+    def __init__(self, u, select=None, msd_type='xyz', fft=True, **kwargs):
+        r"""
+        Parameters
+        ----------
+        u : Universe
+            An MDAnalysis :class:`Universe`.
+        selection : str
+            An MDAnalysis selection string. Defaults to `None` in which case 
+            all atoms are selected.
+        msd_type : {'xyz', 'xy', 'yz', 'xz', 'x', 'y', 'z'}
+            Desired dimensions to be included in the MSD. Defaults to 'xyz'.
+        fft : bool
+            If ``True``, uses a fast FFT based algorithm for computation of 
+            the MSD. Otherwise, use the simple "windowed" algorithm.
+            Defaults to ``True``.
+
+
+        """
+        self.u = u
+
+        super(EinsteinMSD, self).__init__(self.u.trajectory, **kwargs)
+
+        # args
+        self.select = select
+        self.msd_type = msd_type
+        self._parse_msd_type()
+        self.fft = fft
+
+        # local
+        self._atoms = self.u.select_atoms(self.select)
+        self._n_particles = len(self._atoms)
```

Comment:
You could make it a fully documented attribute
```suggestion
        self.n_particles = len(self._atoms)
```
It might be of interest to the user to know how many particles went into the calculation. It also makes your code look more "symmetrical" in relation to `self.n_frames` (see other suggestion).

Your docs also already advertise `n_particles`...